restart_plugin/dmtcp_restart_plugin.cpp

```diff
@@ -39,30 +40,22 @@ void dmtcp_restart_plugin(const string &restartDir,
   // Also, create the DMTCP shared-memory area.
   t->initialize();
 
-  vector<char *> mtcpArgs = getMtcpArgs();
-  mtcpArgs.push_back((char *)"--mpi");
-
-  const map<string, string> &kvmap = t->getKeyValueMap();
-
-  mtcpArgs.push_back((char*) "--minLibsStart");
-  mtcpArgs.push_back((char*) kvmap.at("MANA_MinLibsStart").c_str());
-
-  mtcpArgs.push_back((char*) "--maxLibsEnd");
-  mtcpArgs.push_back((char*) kvmap.at("MANA_MaxLibsEnd").c_str());
-
-  mtcpArgs.push_back((char*) "--minHighMemStart");
-  mtcpArgs.push_back((char*) kvmap.at("MANA_MinHighMemStart").c_str());
+  vector<string> mtcpEnvironment = getMtcpEnvironment(t);
```

Comment:
I see that `getMtcpEnvironment()` is defined in PR #1078 of DMTCP (https://github.com/dmtcp/dmtcp/pull/1078).  As commented there, I think this is too much non-local design.
    Since mtcp_restart is going to throw away any extraneous environment variables when it restores the checkpoint image, let's just pass the _entire_ environment to 'mtcp_restart' through the global `environ` variable.   'mtcp_restart' is going to add special handling for the special environment variables in the 'mtcp' directory.  There's no reason to spread that logic to `restart_plugin/mtcp_restart_plugin.cpp` and to `DMTCP:src/dmtcp_restart.cpp`.
    In the worst case, if you feel that `environ` is not suitable for some reason (please document with a comment), then we should just define `getEnvironment()`.  But I think `environ` should suffice.
    Separately, I assume that this PR will add a commit to include the latest DMTCP, which depends on first finishing PR #1078 of the DMTCP repo.